src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -1245,6 +1352,373 @@ void PDPServer2::send_announcement(
+    uint32_t length;
+    fastrtps::rtps::CacheChange_t* change_aux;
+
+    try
+    {
+        // Create every participant change. If it is external creates it from Reader, 
+        // if it is from the server, it is created from the writer
+        for (auto it = j["participants"].begin(); it != j["participants"].end(); ++it)
+        {
+            length = it.value()["change"]["serialized_payload"]["length"].get<std::uint32_t>();
+            (std::istringstream) it.value()["change"]["sample_identity"].get<std::string>() >> sample_identity_aux;
+
+            // Belongs to own server
+            if (sample_identity_aux.writer_guid() == mp_PDPWriter->getGuid())
+            {
+                if (!mp_PDPWriterHistory->reserve_Cache(&change_aux, length))
+                {
+                    logError(RTPS_PDP_SERVER, "Error creating CacheChange");
+                    // TODO release changes and exit
+                }
+            }
+            // It came from outside
+            else
+            {
+                if (!mp_PDPReaderHistory->reserve_Cache(&change_aux, length))
+                {
+                    logError(RTPS_PDP_SERVER, "Error creating CacheChange");
+                    // TODO release changes and exit
+                }
+            }
+
+            // Deserialize from json to change already created
+            ddb::from_json(it.value()["change"], *change_aux);
+
+            // Insert into the map so the DDB can store it
+            changes_map.insert(
+                    std::make_pair(change_aux->instanceHandle, change_aux));
+
+            // If the change was read as is_local we must pass it to listener with his own writer_guid
+            if (it.value()["is_local"].get<bool>() &&
+                    change_aux->write_params.sample_identity().writer_guid().guidPrefix !=
+                        mp_PDPWriter->getGuid().guidPrefix &&
+                    change_aux->kind == fastrtps::rtps::ALIVE)
+            {
+                change_aux->writerGUID = change_aux->write_params.sample_identity().writer_guid();
+                change_aux->sequenceNumber = change_aux->write_params.sample_identity().sequence_number();
+                mp_listener->onNewCacheChangeAdded(mp_PDPReader, change_aux);
+            }
+        }
+
+        // Create every writer change. If it is external creates it from Reader, 
+        // if it is from the server, it is created from writer
+        for (auto it = j["writers"].begin(); it != j["writers"].end(); ++it)
+        {
+            fastrtps::rtps::CacheChange_t* change_aux;
+            length = it.value()["change"]["serialized_payload"]["length"].get<std::uint32_t>();
+            (std::istringstream) it.value()["change"]["sample_identity"].get<std::string>() >> sample_identity_aux;
+
+            if (it.value()["topic"] == discovery_db().virtual_topic())
+            {
+                fastrtps::rtps::CacheChange_t* change_aux = new fastrtps::rtps::CacheChange_t();
```

Comment:
Duplicate declaration of `change_aux` which raise a warning (first declaration in line 1466).